code/game/objects/effects/decals/Cleanable/fuel.dm

```diff
@@ -154,4 +154,12 @@
 					other_foam.Spread(exclude)
 			else
 				new/obj/effect/decal/cleanable/foam(target, amount*0.25,1)
-			amount *= 0.75
\ No newline at end of file
+			amount *= 0.75
+
+/obj/effect/decal/cleanable/foam/Crossed(AM as mob|obj)
```

Comment:
Not too sure if that is a good idea since that could just result in people walking in circles on extinguishing foam.
Imho it would more sense to implement that in the drop and roll thing you can do when on fire (make it more effective if you do it on a tile covered in foam

Reply:
I thought about same thing. But I want to keep this along with drop and roll. As irrc foam is thick where you would definitely get yourself covered in it by just walking. I think I will make drop and roll suggestion, decease effectiveness of walking lower then drop and roll. What do you think?

Reply:
The problem I see with just walking through foam is that it is a very passive action, that isnt really disrupting. (Which I believe it should be)

Where as drop and roll is a rather active and disrupting action.
You pretty much have to stop doing whatever you are doing to take care of the fire.